tests/test_emitter.py

```diff
@@ -131,27 +168,37 @@ def test_create_wrong_encoding():
 def test_delete():
     touch(p('a'))
     start_watching()
+
     rm(p('a'))
 
-    event = event_queue.get(timeout=5)[0]
-    assert event.src_path == p('a')
-    assert isinstance(event, FileDeletedEvent)
+    # FIXME fseventsd fools the emitter by sending 0x10700 which is a file that is created, modified, and deleted
+    if platform.is_darwin():
+        expect_event(FileModifiedEvent(p('a')))
+
+    expect_event(FileDeletedEvent(p('a')))
 
     if not platform.is_windows():
-        event = event_queue.get(timeout=5)[0]
-        assert os.path.normpath(event.src_path) == os.path.normpath(p(''))
-        assert isinstance(event, DirModifiedEvent)
+        expect_event(DirModifiedEvent(p()))
 
 
 @pytest.mark.flaky(max_runs=5, min_passes=1, rerun_filter=rerun_filter)
 def test_modify():
-    touch(p('a'))
     start_watching()
     touch(p('a'))
 
-    event = event_queue.get(timeout=5)[0]
-    assert event.src_path == p('a')
-    assert isinstance(event, FileModifiedEvent)
+    if platform.is_darwin():
+        expect_event(FileCreatedEvent(p('a')))
+        expect_event(DirModifiedEvent(p()))
+        expect_event(FileModifiedEvent(p('a')))
+
```

Comment:
I thought the idea here was to create 'a' before starting the watch and only modifying it afterwards. This fixes the test failure on Linux where an unexpected `FileCreatedEvent(p('a'))` is emitted.